cli/central-client/src/main/java/org/ballerinalang/central/client/CentralAPIClient.java

```diff
@@ -1022,4 +1021,28 @@ public JsonObject getTrigger(String id, String supportedPlatform, String balleri
             }
         }
     }
+
+    /**
+     * Handle unauthorized response.
+     *
+     * @param org  org name
+     * @param body response body
+     * @throws IOException            when accessing response body
+     * @throws CentralClientException with unauthorized error message
+     */
+    private void handleUnauthorizedResponse(String org, Optional<ResponseBody> body)
+            throws IOException, CentralClientException {
+        if (body.isPresent()) {
+            Optional<MediaType> contentType = Optional.ofNullable(body.get().contentType());
+            if (contentType.isPresent() && isApplicationJsonContentType(contentType.get().toString())) {
+                Error error = new Gson().fromJson(body.get().string(), Error.class);
+                throw new CentralClientException("unauthorized access token for organization: '" + org +
```

Comment:
The token can be a revoked token or an expired token, Can we throw the actual token error?

Reply:
@suganyasuven When I checked the response I didn't see any error description in the response. Can you pls show me the API documentation, So I can do changes according to that?

Reply:
@pramodya1994 We are returning 401 with an error specifying provided token is invalid. However, `unauthorized access token for organization` will mislead the user.

Reply:
Currently, we are not getting an error message from the central API response. That is why we are throwing 2 errors with the reason from central API response and without the reason.

We are showing the error with the relevant organization because an access token is always issued for a given organization.
@hevayo @azinneera WDYT?

Reply:
@pramodya1994 We will add the exact error with the response.

Reply:
Currently, I'm not getting any error from the central API for 401, AFAIK First we have to add the error from the gateway.